app/controllers/ItemShortageOrExcessController.scala

```diff
@@ -68,34 +67,46 @@ class ItemShortageOrExcessController @Inject()(
         getCnCodeInformationService.getCnCodeInformationWithMovementItems(Seq(item)).flatMap {
           serviceResult =>
             val unitOfMeasure = serviceResult.head._2.unitOfMeasureCode.toUnitOfMeasure
+            val isPartiallyRefusingAnAmountOfItem = request.userAnswers.get(RefusingAnyAmountOfItemPage(idx)).contains(true) &&
+              request.userAnswers.get(AcceptMovementPage).contains(PartiallyRefused)
+
             submitAndTrimWhitespaceFromTextarea(Some(ItemShortageOrExcessPage(idx)), formProvider)(
               formWithErrors =>
                 Future.successful(BadRequest(renderView(formWithErrors, ern, arc, idx, mode, unitOfMeasure)))
             ) {
-              case value if (value.wrongWithItem == Shortage) && (value.amount > maxAmount(idx, item)) =>
-                val formWithError =
-                  formProvider()
+              case value if (value.wrongWithItem == Shortage) && (value.amount > maxAmount(idx, item, isPartiallyRefusingAnAmountOfItem)) =>
+                if (isPartiallyRefusingAnAmountOfItem) {
+                  val formWithError = formProvider()
+                    .fill(value)
+                    .withError(FormError(
+                      key = "amount",
+                      message = "itemShortageOrExcess.amount.error.shortageExceedsRemainder",
+                      args = Seq(value.amount, maxAmount(idx, item, isPartiallyRefusingAnAmountOfItem))
```

Comment:
Where it currently uses: `value.amount` this needs to change. It should be the `RefusedAmount` that is passed into this.

Reply:
I also wonder if the `UnitOfMeasure` needs to be passed to the view. Might be worth checking with clare, as this doesn't really make sense to me:
![image](https://github.com/hmrc/emcs-tfe-report-a-receipt-frontend/assets/16954819/f33dba20-6367-403a-90e2-a2b5cb283674)

Saying:
> "You have refused 1 litre of this item. The amount of shortage cannot be more than 0"
> "You have refused 2 litres of this item. The amount of shortage cannot be more than 0"
> "You have refused 10 kilograms of this item. The amount of shortage cannot be more than 45"
etc...

Might make more sense? 🤷